File: backend/GraficaModerna.Application/Interfaces/IProductService.cs
using GraficaModerna.Application.DTOs;

namespace GraficaModerna.Application.Interfaces;

public interface IProductService
{
    Task<PagedResultDto<ProductResponseDto>> GetCatalogAsync(string? search, string? sort, string? order, int page,
        int pageSize);

    Task<ProductResponseDto> GetByIdAsync(Guid id);
    Task<ProductResponseDto> CreateAsync(CreateProductDto dto);
    Task UpdateAsync(Guid id, UpdateProductDto dto);
    Task DeleteAsync(Guid id);
}

File: grafica-moderna/app/Application/IProductService.php
<?php

namespace App\Application\Interfaces;

interface IProductService
{
    /**
     * @return array{items: array, total: int, page: int, pageSize: int}
     */
    public function getCatalog(
        ?string $search, 
        ?string $sort, 
        ?string $order, 
        int $page,
        int $pageSize
    ): array;

    public function getById(string $id): ?object;

    public function create(array $dto): object;

    public function update(string $id, array $dto): void;

    public function delete(string $id): void;
}